### PR TITLE
Init template

### PR DESCRIPTION
**Pull request by dhuckins**
allows `ent init ...` to accept template string/file for new model creation

### CODE REVIEW

File: cmd/internal/base/base.go [hunk 1]
```diff
@@ -65,12 +65,15 @@ func (IDType) String() string {
...
 	var target string
+	var newSchemaTemplate string
 	cmd := &cobra.Command{
 		Use:   "init [flags] [schemas]",
 		Short: "initialize an environment with zero or more schemas",
 		Example: examples(
 			"ent init Example",
 			"ent init --target entv1/schema User Group",
+			"ent init --template '<template string>'",
+			"ent init --template file=path/to/tmpl.tmpl",
```
Comment (a8m) [inline cmd/internal/base/base.go:76]:
```suggestion
			"ent init --template file=path/to/file.tmpl User",
```

Comment (a8m) [reply cmd/internal/base/base.go:76]: Please, let's not support string template atm. It's not supported by `ent generate` and I don't see a use case for it.

File: cmd/internal/base/base.go [hunk 2]
```diff
@@ -65,12 +65,15 @@ func (IDType) String() string {
 // InitCmd returns the init command for ent/c packages.
 func InitCmd() *cobra.Command {
 	var target string
+	var newSchemaTemplate string
```
Comment (a8m) [inline cmd/internal/base/base.go:68]:
```suggestion
	var target, tmpl string
```

Keep variable names short please.

**Review by a8m — CHANGES_REQUESTED** [review]
Thanks for the contribution @dhuckins! I actually like this feature a lot 🚀 

Please, see my comment below.

File: cmd/internal/base/base.go (hunk 1 above)
Comment (dhuckins) [reply cmd/internal/base/base.go:76]: thought I saw it elsewhere (?)

took a second look and couldn't find, will remove

File: cmd/internal/base/base.go [hunk 3]
```diff
@@ -65,12 +65,14 @@ func (IDType) String() string {
 // InitCmd returns the init command for ent/c packages.
 func InitCmd() *cobra.Command {
 	var target string
+	var tmplPath string
```
Comment (a8m) [inline cmd/internal/base/base.go:68]:
```suggestion
	var target, tmplPath string
```

File: cmd/internal/base/base.go [hunk 4]
```diff
@@ -65,12 +65,14 @@ func (IDType) String() string {
...
 	cmd := &cobra.Command{
 		Use:   "init [flags] [schemas]",
 		Short: "initialize an environment with zero or more schemas",
 		Example: examples(
 			"ent init Example",
 			"ent init --target entv1/schema User Group",
+			"ent init --template ./path/to/tmpl.tmpl User",
```
Comment (a8m) [inline cmd/internal/base/base.go:75]:
```suggestion
			"ent init --template ./path/to/file.tmpl User",
```

File: cmd/internal/base/base.go [hunk 5]
```diff
@@ -81,12 +83,26 @@ func InitCmd() *cobra.Command {
...
+				tmpl, err = template.ParseFiles(tmplPath)
+			} else {
+				tmpl, err = template.New("schema").Parse(defaultTemplate)
+			}
+
+			if err != nil {
+				log.Fatalln(fmt.Errorf("ent/init: could not compile template %w", err))
+			}
+
+			if err := initEnv(target, names, tmpl); err != nil {
```
Comment (a8m) [inline cmd/internal/base/base.go:99]:
```suggestion
			var (
				err error
				tmpl *template.Template
			)
			if tmplPath != "" {
				tmpl, err = template.ParseFiles(tmplPath)
			} else {
				tmpl, err = template.New("schema").Parse(defaultTemplate)
			}
			if err != nil {
				log.Fatalln(fmt.Errorf("ent/init: could not parse template %w", err))
			}
			if err := initEnv(target, names, tmpl); err != nil {
```

**Review by a8m — APPROVED** [review]
💯 

Looks good. Please address the minor comments and let's merge it.

**Comment by dhuckins** [comment]
@a8m suggestions applied! needs an approval from a contributor to run CI checks

thanks!

**Review by a8m — APPROVED** [review]
I love this feature a lot! Thanks again for adding this ❤️

**Comment by a8m** [comment]
Please, you are welcome to join our [Discord community](https://discord.gg/qZmPgTE6RX).

**Pull request merged**